docker/jupyter/custom/jupyter_delocalize.py

```diff
@@ -178,21 +178,30 @@ def __init__(self, *args, **kwargs):
     self.welder_base_url = 'http://127.0.0.1:8080'
     super(WelderContentsManager, self).__init__(*args, **kwargs)
 
-  def _welder_delocalize(self, path):
+  def _post_welder(self, action, path):
     # Ignore storage link failure, throw other errors.
     resp = requests.post(self.welder_base_url + '/objects', data=json.dumps({
-      'action': 'safeDelocalize',
+      'action': action,
       # Sometimes the Jupyter UI provided "path" contains a leading /, sometimes
       # not; strip for Welder.
       'localPath': path.lstrip('/')
     }))
     if not resp.ok:
-      msg = resp.reason or 'unknown Welder error'
+      error_json = {}
       try:
-        msg = json.dumps(resp.json())
+        error_json = resp.json()
       except:
         pass
-      raise IOError("safeDelocalize failed: " + msg)
+
+      if resp.status_code == 412 and error_json.get('errorCode', -1) == 1:
+        # 1: Storage Link not found; expected for unmanaged files.
+        return
+
+      msg = resp.reason or 'unknown Welder error'
+      if error_json != {}:
+        msg = json.dumps(error_json)
```

Comment:
just curious, do error messages get logged somewhere?

Reply:
I see them getting logged by Tornado / Jupyter locally, but worth confirming once we're able to deploy this fully.